File: LinkedLists/delete_the_middle_node_of_a_linked_list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteMiddle(ListNode* head) {
        int size = 0;
        ListNode* temp = head;
        while(temp!=NULL){
            size++;
            temp = temp->next;
        }
        
        if(size<2){
            head = head->next;
            return head;
        }
        
        int n = (size/2)-1;
        
        ListNode* index = head;
        while(n>0){
            index = index->next;
            n--;
        }
        
        index->next = index->next->next;
        
        return head;
    }
};

File: LinkedLists/delete_the_middle_node_of_a_linked_list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteMiddle(self, head: Optional[ListNode]) -> Optional[ListNode]:
        temp = head
        size=0
        while(temp!=None):
            size+=1;
            temp = temp.next
        if size==1:
            head = head.next
            return head
        if size%2==0:
            n = (size/2)
        else:
            n = (size/2)-1
        index = head
        while(n-1>0):
            n-=1
            index = index.next
        index.next = index.next.next
        return head